scipy/stats/tests/test_entropy.py

```diff
@@ -1,111 +1,129 @@
-import numpy as np
-from numpy.testing import assert_equal, assert_allclose
-# avoid new uses of the following; prefer assert/np.testing.assert_allclose
-from numpy.testing import (assert_, assert_almost_equal,
-                           assert_array_almost_equal)
-
+import math
 import pytest
 from pytest import raises as assert_raises
-import scipy.stats as stats
 
+import numpy as np
+from numpy.testing import assert_allclose
+
+from scipy import stats
+from scipy.conftest import array_api_compatible
+from scipy._lib._array_api import xp_assert_close, xp_assert_equal, xp_assert_less
 
 class TestEntropy:
-    def test_entropy_positive(self):
+    @array_api_compatible
+    def test_entropy_positive(self, xp):
         # See ticket #497
-        pk = [0.5, 0.2, 0.3]
-        qk = [0.1, 0.25, 0.65]
+        pk = xp.asarray([0.5, 0.2, 0.3])
+        qk = xp.asarray([0.1, 0.25, 0.65])
         eself = stats.entropy(pk, pk)
         edouble = stats.entropy(pk, qk)
-        assert_(0.0 == eself)
-        assert_(edouble >= 0.0)
+        xp_assert_equal(eself, xp.asarray(0.))
+        xp_assert_less(-edouble, xp.asarray(0.))
 
-    def test_entropy_base(self):
-        pk = np.ones(16, float)
+    @array_api_compatible
+    def test_entropy_base(self, xp):
+        pk = xp.ones(16)
         S = stats.entropy(pk, base=2.)
-        assert_(abs(S - 4.) < 1.e-5)
+        xp_assert_less(xp.abs(S - 4.), xp.asarray(1.e-5))
 
-        qk = np.ones(16, float)
-        qk[:8] = 2.
+        qk = xp.ones(16)
+        qk = xp.where(xp.arange(16) < 8, xp.asarray(2.), qk)
         S = stats.entropy(pk, qk)
         S2 = stats.entropy(pk, qk, base=2.)
-        assert_(abs(S/S2 - np.log(2.)) < 1.e-5)
+        xp_assert_less(xp.abs(S/S2 - math.log(2.)), xp.asarray(1.e-5))
 
-    def test_entropy_zero(self):
+    @array_api_compatible
+    def test_entropy_zero(self, xp):
         # Test for PR-479
-        assert_almost_equal(stats.entropy([0, 1, 2]), 0.63651416829481278,
```

Comment:
Every line is changed, so I suppose `git` can't tell, but it's a very straightforward conversion. Consider using a diff tool locally.